src/jvmMain/kotlin/com/sdercolin/vlabeler/video/VideoCore.kt

```diff
@@ -0,0 +1,60 @@
+// see https://github.com/JetBrains/compose-jb/pull/1738 for direct control of awt/swing
+@file:Suppress("INVISIBLE_REFERENCE", "INVISIBLE_MEMBER")
+
+package com.sdercolin.vlabeler.video
+import androidx.compose.foundation.layout.Box
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.DisposableEffect
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.awt.LocalLayerContainer
+import androidx.compose.ui.geometry.Offset
+import androidx.compose.ui.layout.onGloballyPositioned
+import androidx.compose.ui.platform.LocalDensity
+import androidx.compose.ui.unit.round
+import java.awt.BorderLayout
+import java.awt.Container
+import javax.swing.JPanel
+
+private lateinit var container: Container
```

Comment:
Please move it into the composable. You can create a simplified version of `ComponentInfo` which is used in `androidx.compose.ui.awt.SwingPanel`
```kotlin

private class ComponentInfo {
   lateinit var container: Container
}

fun VideoCore(videoState: VideoState, modifier: Modifier) {
   val componentInfo = remember { ComponentInfo() }
   ..
   DisposableEffect(Unit) {
        componentInfo.container = JPanel().apply {
        ..
    }
...
```

Reply:
That would be better. Compose ui's official code is a good place to learn things.